lib/db/mem.js

```diff
@@ -380,6 +380,33 @@ module.exports = function (log, error) {
       )
   }
 
+  Memory.prototype.sessionDevice = function (id) {
+    return this.sessionToken(id)
+      .then(
+        function (session) {
+          return this.accountDevices(session.uid)
+            .then(
+              function (devices) {
+                var device = devices.filter(
+                  function (d) {
+                    return d.sessionTokenId.toString('hex') === id.toString('hex')
+                  }
+                )[0]
+                if (device) {
+                  session.deviceId = device.id
+                  session.deviceName = device.name
+                  session.deviceType = device.type
+                  session.deviceCreatedAt = device.createdAt
+                  session.callbackURL = device.callbackURL
+                  session.callbackPublicKey = device.callbackPublicKey
```

Comment:
I don't feel strongly either way, but out of curiosity, why not attach the device as a sub-object like `session.device = device`?